lib/jerakia/client.rb

```diff
@@ -93,22 +99,31 @@ def http_send(request)
       end
     end
 
+    def set_content_type(request)
+      if not @config.key?(:content_type) or @config[:content_type] == 'json'
+        request.add_field('Content-Type', 'application/json')
+        content_method = 'to_json'
+      else @config[:content_type] == 'msgpack'
+        request.add_field('Content-Type', 'application/x-msgpack')
+        content_method = 'to_msgpack'
+      end
+      return request
+    end
 
     def get(url_path, params={})
       uri = URI.parse(url_address +  url_path)
       uri.query = URI.encode_www_form(params)
       request = Net::HTTP::Get.new(uri.to_s)
+      request = set_content_type(request)
       return http_send(request)
     end
 
     def put(url_path, params)
       uri = URI.parse(url_address + url_path)
       request = Net::HTTP::Put.new(uri.path)
-      request.add_field('Content-Type', 'application/json')
-      request.body = params.to_json
+      request = set_content_type(request)
+      request.body = params.method(@method_name).call
```

Comment:
See the comment on your PR in Jerakia core, couldnt we do it the same way here?  `request.body = encode_params(params)` or something?